src/cpp/rtps/builtin/discovery/database/DiscoverySharedInfo.cpp

```diff
@@ -0,0 +1,40 @@
+// Copyright 2020 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file DiscoverySharedInfo.cpp
+ *
+ */
+
+#include <fastdds/rtps/common/CacheChange.h>
+#include <fastdds/rtps/common/GuidPrefix_t.hpp>
+
+#include "./DiscoverySharedInfo.hpp"
+
+namespace eprosima {
+namespace fastdds {
+namespace rtps {
+namespace ddb {
+
+void DiscoverySharedInfo::set_disposal(
+        eprosima::fastrtps::rtps::CacheChange_t* change)
+{
+    relevant_participants_builtin_ack_status_.match_all();
+    change_info(change);
```

Comment:
Should we check whether the changes is a disposal?

Reply:
no, that is done in process_data_queue. If there is not a disposal it should not call `set_disposal`